contracts/src/Gateway.sol

```diff
@@ -617,4 +621,31 @@ contract Gateway is IGateway, IInitializable {
         assets.assetHubCreateAssetFee = config.assetHubCreateAssetFee;
         assets.assetHubReserveTransferFee = config.assetHubReserveTransferFee;
     }
+
+    /// @inheritdoc IGateway
+    function transact(
+        ParaID destinationChain,
+        OriginKind originKind,
+        uint128 destinationFee,
+        Weight calldata weightAtMost,
+        bytes calldata call
+    ) external payable {
+        if (call.length < 2 || destinationFee == 0 || weightAtMost.refTime == 0 || weightAtMost.proofSize == 0) {
+            revert InvalidTransact();
+        }
```

Comment:
I feel a bit uncomfortable having the source chain interpreting and validating these fields - I think that's the job of the destination chain. Or in other words, I think we can pass these values through safely without validation.

Reply:
https://github.com/Snowfork/snowbridge/commit/00abc1330b4eb6e0932ddbc7e8d1bbe1ca757eae

Do you think better to add some native costs here to avoid spamming?